src/main/java/com/glencoesoftware/ldaptool/Main.java

```diff
@@ -112,58 +123,81 @@ public GroupLoader newGroupLoader(
         String referral = context.getProperty("omero.ldap.referral");
         Field ignorePartialResultException =
-            LdapTemplate.class.getDeclaredField("ignorePartialResultException");
+                LdapTemplate.class.getDeclaredField("ignorePartialResultException");
         ignorePartialResultException.setAccessible(true);
         log.info("Ignoring partial result exceptions? {}",
                 ignorePartialResultException.get(ldapTemplate));
         log.info("Referral set to: '{}'", referral);
-        String dn = ldapImpl.findDN(username);
-        log.info("Found DN: {}", dn);
-        Experimenter experimenter = ldapImpl.findExperimenter(username);
-        log.info(
-            "Experimenter field mappings id={} email={} firstName={} " +
-            "lastName={} institution={} ldap={} middleName={} omeName={}",
-            experimenter.getId(), experimenter.getEmail(),
-            experimenter.getFirstName(), experimenter.getLastName(),
-            experimenter.getInstitution(), experimenter.getLdap(),
-            experimenter.getMiddleName(), experimenter.getOmeName()
-        );
 
+        System.out.println("---");
+        if (searchFor.all) {
+            lookupAllUsers(ldapImpl, ldapTemplate);
+        } else {
+            try {
+                Experimenter experimenter = ldapImpl.findExperimenter(searchFor.username);
+                lookupUser(ldapImpl, ldapTemplate, experimenter);
+            } catch (ome.conditions.ApiUsageException api) {
+                System.err.println("no such user: " + searchFor.username);
+                return 1;
+            }
+        }
+        return 0;
+    }
+
+    public void lookupAllUsers(LdapImpl ldapImpl, LdapTemplate ldapTemplate) throws Exception {
+        List<Experimenter> users = ldapImpl.searchAll();
+        for (Experimenter user : users) {
+            lookupUser(ldapImpl, ldapTemplate, user);
+        }
+    }
+
+    public void lookupUser(LdapImpl ldapImpl, LdapTemplate template, Experimenter user) throws Exception {
+
+        String dn = (String) user.retrieve("LDAP_DN");
+
+        // This class needs updating in omero-server to make it also return strings
         GroupLoader groupLoader = newGroupLoader(
-                ldapImpl, username, new DistinguishedName(dn));
+                ldapImpl, user.getOmeName(), new DistinguishedName(dn));
         Field groups = LdapImpl.GroupLoader.class.getDeclaredField("groups");
         groups.setAccessible(true);
-        List<Long> groupIds = (List<Long>) groups.get(groupLoader);
-        List<Long> ownedGroupIds = groupLoader.getOwnedGroups();
-        log.info(
-            "Would be member of Group IDs={}",
-            Arrays.toString(groupIds.toArray())
-        );
-        log.info(
-            "Would be owner of Group IDs={}",
-            Arrays.toString(ownedGroupIds.toArray())
-        );
+        System.out.println("- dn: " + dn);
```

Comment:
These really should be in quotes. Any objections?

Reply:
None. Completely up to you.